web/src/pages/NewPlayerPage.js

```diff
@@ -0,0 +1,34 @@
+import AddPlayerForm from "../components/AddPlayerForm";
+
+function NewPlayerPage() {
+    function addPlayerHandler(playerData) {
+        console.log("------", playerData)
+        // send HTTP request to fetched URL, where you want to send
+        fetch(
```

Comment:
By looking at the code, you seemed to try taking the user to the successful page after the registration. 
To do that, we need `async/await` on `fetch. 

1. change the function as `async` function.
before
```
function addPlayerHandler(playerData) {
// or
const addPlayerHandler = (playerData) => {
```
after
```
async function addPlayerHandler(playerData) {
// or 
const addPlayerHandler = async (playerData) => {
```
Notice where to add `async`. 

2. add `await` on `fetch`
```
try {
  await fetch(// the same);
  // Successful. Redirect the user.
} catch (e) {
  // Something wrong happened, show some error message or alert. 
}
```
This way, we can add some post-logics "after" the fetch. Notice that we need `await` here because the code needs to wait until the network call has resulted. 

note: We can do the same with `Promise` syntax, which is a bit more complex.

2-2. Send a successful result from the server
Even after adding `await` on fetch as above, the code won't work and the program won't come back to the place where after `await`. It's because our `/admin/registration` route is not sending a successful message to the front end. 

We can add `res.send("ok");` on the server line as the last step of `/admin/registration` route. Then this `await` will work correctly.

3. We can now take the user to the new page when the registration is successful. 
```
window.location = "/registered";
```

Reply:
How can I check if the alert will be shown correctly, when an error is hatched?

Reply:
We can use the wrong URL that doesn't exist on purpose. It will go to the `catch`.

Reply:
Sorry, it was my bad. There was a mistake in my code. 
```
try {
      const fetchedResult = await fetch(...);
      const response = await fetchedResult.json();
      if (response.status === 200) {
        // SUCCESSFUL!
      } else {
        // NETWORK CALL FAILED (payload was not valid or something)
      }
    } catch (e) {
      // The usage of fetch was something wrong OR json data from server was invalid
    }
```
Notice that we will go to the three scenarios above. 

2. also mistake for server code.
not `res.send("ok");` but we need to return something `json` because frontend expects to receive `json.
```
res.send({});
// or
res.send({additionalInfo: "good job"});
```
Whatever it is, we need to send `json`(object) not `string`.

Reply:
There was a typo. I fixed it in the line. If you checked my updated comment, never mind.
```
if (fetchResult.status === 200) {
```